python/mrc/_pymrc/tests/test_asyncio_runnable.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2021-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "test_pymrc.hpp"
+
+#include "pymrc/asyncio_runnable.hpp"
+#include "pymrc/executor.hpp"
+#include "pymrc/pipeline.hpp"
+#include "pymrc/types.hpp"
+
+#include "mrc/coroutines/async_generator.hpp"
+#include "mrc/node/rx_sink.hpp"
+#include "mrc/node/rx_source.hpp"
+#include "mrc/options/engine_groups.hpp"
+#include "mrc/options/options.hpp"
+#include "mrc/options/topology.hpp"
+#include "mrc/runnable/types.hpp"
+#include "mrc/segment/builder.hpp"
+#include "mrc/segment/object.hpp"
+
+#include <boost/fiber/policy.hpp>
+#include <gtest/gtest.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/pytypes.h>
+#include <rxcpp/rx.hpp>
+
+#include <atomic>
+#include <coroutine>
+#include <memory>
+#include <string>
+
+namespace py    = pybind11;
+namespace pymrc = mrc::pymrc;
+using namespace std::string_literals;
+using namespace py::literals;
+
+PYMRC_TEST_CLASS(AsyncioRunnable);
+
+class MyAsyncioRunnable : public pymrc::AsyncioRunnable<int, unsigned int>
+{
+    mrc::coroutines::AsyncGenerator<unsigned int> on_data(int&& value) override
+    {
+        co_yield value;
+        co_yield value * 2;
+    };
+};
+
```

Comment:
I feel like we could do with a more robust set of tests, and more tests for each of the underlying components.

Reply:
I've added some more test cases, but the granularity is not great due to the nature of the MRC launch infrastructure. Generally happy path and exception paths are tested.